gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -583,4 +604,50 @@ public Completable insertOrUpdateOfflineArea(OfflineArea area) {
         .toSingle()
         .subscribeOn(schedulers.io());
   }
+
+  private Completable deleteTiles(OfflineAreaEntity offlineAreaEntity) {
+    return tileSourceDao
+        .getTileSourcesWithOfflineAreas()
+        .toSingle()
+        .map(
+            sources ->
+                stream(sources)
+                    .filter(
+                        source ->
+                            // We only remove tiles when the given area is the only one that depends
+                            // on them.
+                            source.offlineAreaEntities.contains(offlineAreaEntity)
+                                && source.offlineAreaEntities.size() == 1)
+                    .map(source -> source.tileSourceEntity)
+                    .collect(toImmutableList()))
+        .flatMapObservable(Observable::fromIterable)
+        .flatMapCompletable(
+            tile ->
+                // TODO: Ideally, we'll want to decouple the file deletion from the local data store
+                Completable.fromAction(() -> fileUtil.deleteFile(tile.getPath()))
+                    .andThen(
+                        tileSourceDao
+                            .delete(tile)
+                            .andThen(
+                                offlineAreaTileSourceCrossRefDao.delete(
+                                    OfflineAreaTileSourceCrossRef.builder()
+                                        .setOfflineAreaId(offlineAreaEntity.getId())
+                                        .setTileSourcePath(tile.getPath())
+                                        .build()))))
+        .doOnComplete(() -> Timber.d("Removed tiles"))
+        .doOnError(throwable -> Timber.e(throwable, "Failed to delete tiles"));
```

Comment:
Imo logging should be done in the subscriber (where possible) to avoid duplicate logging messages - lmk what you think!